Source/ASDisplayNode.mm

```diff
@@ -729,31 +729,39 @@ - (BOOL)_locked_isNodeLoaded
 
 - (UIView *)view
 {
-  ASDisplayNodeAssert(!_flags.layerBacked, @"Call to -view undefined on layer-backed nodes");
-  if (_flags.layerBacked) {
-    return nil;
-  }
-
-  if (_view == nil) {
-    ASDisplayNodeAssertMainThread();
-    [self _loadViewOrLayerIsLayerBacked:NO];
+  {
+    ASDN::MutexLocker l(__instanceLock__);
+    ASDisplayNodeAssert(!_flags.layerBacked, @"Call to -view undefined on layer-backed nodes");
+    if (_flags.layerBacked) {
+      return nil;
+    }
+    
+    if (_view != nil) {
+      return _view;
+    }
   }
 
+  ASDisplayNodeAssertMainThread();
+  [self _loadViewOrLayerIsLayerBacked:NO];
```

Comment:
Don't we need the lock while loading view? Otherwise, if a background thread calls `isNodeLoaded`, it will get a no while the view is literally being loaded.

So this maybe:
```objective-c
- (UIView *)view
{
  ASDN::MutexLocker l(__instanceLock__);
  ASDisplayNodeAssert(!_flags.layerBacked, @"Call to -view undefined on layer-backed nodes");
  if (_flags.layerBacked) {
    return nil;
  }
 
  if (_view != nil) {
    return _view;
  }
  
  ASDisplayNodeAssertMainThread();
  [self _loadViewOrLayerIsLayerBacked:NO]; 
  return _view;
}
```